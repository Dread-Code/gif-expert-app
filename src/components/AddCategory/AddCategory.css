.add-category-container {
  display: flex;
}

*:focus {
  outline: none;
}

input {
  justify-self: center;
  font-size: 1.3rem;
  width: 100%;
  color: grey;
  border-radius: 0.4em;
  margin: 0.5em;
}
