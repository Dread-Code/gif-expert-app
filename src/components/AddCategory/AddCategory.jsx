import React, { useState } from "react";
import "./AddCategory.css";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    setCategories(inputValue);
  };

  return (
    <form className="add-category-container" onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder={"Que quieres buscar?"}
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />
    </form>
  );
};

export default AddCategory;
