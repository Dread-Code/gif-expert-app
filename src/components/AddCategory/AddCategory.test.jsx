import { shallow } from 'enzyme'
import AddCategory from './AddCategory'

describe('AddCAtegory Tests', () => {
  const setCategories = jest.fn()
  const categories = []
  let wrapper = shallow(<AddCategory setCategories={setCategories} categories={categories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} categories={categories} />)
  })

  test('should show correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should change the text box', () => {
    const input = wrapper.find('input')
    const value = 'Hola Mundo'
    // EL segundo parametro seria la simulacion de e.target.value
    input.simulate('change', { target: { value } })
  })

  test('Should not post the information', () => {
    // recuerda utilizar beforeeach para restaurar el wrapper en cada test(cuando sea necesario)
    wrapper.find('form').simulate('submit', { preventDefault() {} })
    expect(setCategories).not.toHaveBeenCalled()
  })
})
