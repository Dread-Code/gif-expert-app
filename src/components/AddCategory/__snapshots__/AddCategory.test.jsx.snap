// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCAtegory Tests should show correctly 1`] = `
<form
  className="add-category-container"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Que quieres buscar?"
    type="text"
    value=""
  />
</form>
`;
